src/number/format-decimal.js

```diff
@@ -42,7 +51,11 @@ module.exports = function(value, localeData, options) {
 
 		digitNum++;
 
-		if (digitNum > 1 && digitNum % localeData.groupSize === 1) {
+		if (digitNum > 1 && digitNum % currentGroupSize === 1) {
+			if ( groupSizes.length > 0) {
+				currentGroupSize = groupSizes.shift();
+			}
+			digitNum = 1;	// reset this back to 1 to handle variable group sizes
```

Comment:
If you started with digitNum=1, then the counting would be off by one. So you'd get 12,345,67, instead of 1,234,567. 
The algorithm goes right to left, so you want to add commas when digitnum=4 (in this case '4'), and when digitnum=7 (in this case '1')

The confusion comes because now that we have variable group sizes. I could re-factor this to use countdown group counters instead. That might be easier to understand. Let me give that a try and see what you guys think.